docs/DeveloperGuide.md

```diff
@@ -1,34 +1,383 @@
 # Developer Guide
 
+- [Design & implementation](#design--implementation)
+    - [Architecture](#architecture)  
+    - [Ui component](#ui-component)  
+    - [Logic component](#logic-component)  
+    - [Model component](#model-component)  
+    - [Storage component](#storage-component)
+- [Product scope](#product-scope)
+    - [Target user profile](#target-user-profile)
+    - [Value proposition](#value-proposition)
+- [User stories](#user-stories)
+- [Use cases](#use-cases)
+- [Non-functional requirements](#non-functional-requirements) 
+- [Glossary](#glossary)
+- [Instruction for manual testing](#instructions-for-manual-testing)
+
+
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+
+
+![Architecture](diagrams/Architecture.png)
```

Comment:
I would like to suggest to move the description to above the image, just so that the user is introduced to the image before reading them

Reply:
![image](https://user-images.githubusercontent.com/48993548/97530529-4f1a4f80-19ed-11eb-966d-1047f80da59a.png)
I suggest using a double arrow on some of the diagrams as the interaction is both ways. Aside from that, would it be possible that the `Commons` is actually linked to some other part?